web/client/components/misc/toolbar/Toolbar.jsx

```diff
@@ -1,45 +1,46 @@
 /**
- * A Generic Reusable Toolbar. Build in animations.
- * @class Toolbar
- * @memberof components.misc.toolbar
- * @param  {Array}  [buttons=[]]       Array of buttons. Button objects support the following properties:
- *  - *visible*: true by default, set to `false` to make the button disappear
- *  - All the react-bootstrap buttons properties.
- *  - All the properties for @see components.misc.enhancers
- *  . All properties for @see components.misc.toolbar.ToolbarButton and react-bootstrap button
- * @param  {Object} [btnDefaultProps] A series of default Props for buttons
- * @param  {Object} [btnGroupProps] Props to add to the react-bootstrap `ButtonGroup` component
- * @param  {Object|boolean} [transitionProps] properties of ReactCSSTransitionGroup. If false transition are  disabled. (for vertical toolbar)
- */
+* A Generic Reusable Toolbar. Build in animations.
+* @class Toolbar
+* @memberof components.misc.toolbar
+* @param  {Array}  [buttons=[]]       Array of buttons. Button objects support the following properties:
+*  - *visible*: true by default, set to `false` to make the button disappear
+*  - All the react-bootstrap buttons properties.
+*  - All the properties for @see components.misc.enhancers
+*  . All properties for @see components.misc.toolbar.ToolbarButton and react-bootstrap button
+* @param  {Object} [btnDefaultProps] A series of default Props for buttons
+* @param  {Object} [btnGroupProps] Props to add to the react-bootstrap `ButtonGroup` component
+* @param  {Object|boolean} [transitionProps] properties of ReactCSSTransitionGroup. If false transition are  disabled. (for vertical toolbar)
+*/
 module.exports = ({
-    buttons = [],
-    btnGroupProps={},
-    btnDefaultProps = {},
-    transitionProps = {
-        transitionName: "toolbar-btn-transition",
-        transitionEnterTimeout: 300,
-        transitionLeaveTimeout: 300
-    }} = {}) => {
-        const renderButtons = () => buttons.map(
-            ({ visible = true, ...props }, index) => visible
-                ? (<ToolbarButton key={props.key || index} {...btnDefaultProps} {...props} />)
-                : null
-        );
-        return (<ButtonGroup {...btnGroupProps}>
-            {transitionProps
-                ? <ReactCSSTransitionGroup {...transitionProps}>{renderButtons()}</ReactCSSTransitionGroup>
-                : renderButtons()}
-            </ButtonGroup>);
-    };
+   buttons = [],
+   btnGroupProps={},
+   btnDefaultProps = {},
+   transitionProps = {
+       transitionName: "toolbar-btn-transition",
+       transitionEnterTimeout: 300,
+       transitionLeaveTimeout: 300
+   }} = {}) => {
+       const renderButtons = () => buttons.map(
+           ({ visible = true, ...props }, index) => visible
+               ? (isFunction(props.el) && <props.el key={props.key || index} {...props} /> || <ToolbarButton key={props.key || index} {...btnDefaultProps} {...props} />)
```

Comment:
isFunction won't catch any possible component type.  Do you need el in the ToolbarButton use case? If not, you can simply check if props.el is defined here.